python/pyspark/sql/readwriter.py

```diff
@@ -441,7 +441,12 @@ def csv(self, path, schema=None, sep=None, encoding=None, quote=None, escape=Non
                   When it meets a record having fewer tokens than the length of the schema, \
                   sets ``null`` to extra fields. When the record has more tokens than the \
                   length of the schema, it drops extra tokens.
-                * ``DROPMALFORMED`` : ignores the whole corrupted records.
+                * ``DROPMALFORMED`` : ignores the whole corrupted records. Note that when CSV \
+                  parser column pruning (``spark.sql.csv.parser.columnPruning.enabled``) is \
+                  enabled (it is enabled by default), the malformed columns can be ignored during \
+                  parsing if they are pruned, resulting the corrupted records are not dropped. \
+                  Disabling the column pruning feature can drop corrupted records even malformed \
+                  columns are not read.
```

Comment:
I think it's good to note. But for JIRA itself, do you know the cause? I haven't taken a close look but it looked to me like Univocity parses the tokens weirdly in a certain condition. It is possible that it's a bug.

Reply:
If we're going to investigate it separately, let's just file a separate JIRA for this note. Then it should be good to go orthogonally.

Reply:
Not sure which one you think is possibly a bug. Let's discuss it on the JIRA?

Reply:
> If we're going to investigate it separately, let's just file a separate JIRA for this note. Then it should be good to go orthogonally.

Ok. Sounds good.

Reply:
Yea, let's discuss there. For this note, looks good and useful to have. I will take another look tomorrow and get this in